boa3/builtin/compile_time/__init__.py

```diff
@@ -343,8 +344,8 @@ def add_permission(self, *, contract: str = IMPORT_WILDCARD, methods: list[str,
             return
 
         # verifies if methods is a valid value
-        elif ((isinstance(methods, str) and methods != IMPORT_WILDCARD) or
-              (isinstance(methods, list) and any(not isinstance(method, str) for method in methods))):
+        elif ((isinstance(methods, tuple) and (any(not isinstance(method, str) for method in methods) or len(methods) == 0)) or
+              (isinstance(methods, list) and (any(not isinstance(method, str) for method in methods) or len(methods) == 0))):
```

Comment:
the validation is the same for both `tuple` and `list` right?
you could just wright:
```suggestion
        elif ((isinstance(methods, (tuple, list)) and (any(not isinstance(method, str) for method in methods) or len(methods) == 0))):
```